jobbergate-api/jobbergate_api/apps/job_scripts/routers.py

```diff
@@ -63,99 +63,93 @@ async def job_script_create(
     status_code=status.HTTP_201_CREATED,
     response_model=JobScriptResponse,
     description="Endpoint for job script creation",
+    dependencies=[Depends(file_services(file_service, template_file_service))],
 )
 async def job_script_create_from_template(
     create_request: JobScriptCreateRequest,
     render_request: RenderFromTemplateRequest,
     id_or_identifier: int | str = Path(...),
-    session: AsyncSession = Depends(db_session),
-    job_script_service: TableService = Depends(job_script_service),
-    job_script_file_service: FileService = Depends(job_script_files_service),
-    template_service: TableService = Depends(template_service),
-    template_file_service: FileService = Depends(template_files_service),
-    token_payload: TokenPayload = Depends(guard.lockdown(Permissions.JOB_SCRIPTS_EDIT)),
+    secure_session: SecureSession = Depends(
+        secure_services(
+            Permissions.JOB_SCRIPTS_EDIT,
+            services=[crud_service, template_crud_service, file_service, template_file_service],
+        )
+    ),
 ):
     """Create a new job script from a job script template."""
     logger.info(f"Creating a new job script from {id_or_identifier=} with {create_request=}")
 
-    identity_claims = IdentityClaims.from_token_payload(token_payload)
-    if identity_claims.email is None:
+    if secure_session.identity_payload.email is None:
         raise HTTPException(
             status_code=status.HTTP_400_BAD_REQUEST,
             detail="The token payload does not contain an email",
         )
 
-    if not render_request.template_output_name_mapping:
+    base_template = cast(JobScriptTemplate, await template_crud_service.get(id_or_identifier))
+
+    required_map = render_request.template_output_name_mapping
+    entrypoint_key = list(required_map.values())[0]
+    mapped_template_files = {}
+    for template_file in base_template.template_files:
+        new_filename = required_map.pop(template_file.filename, None)
+        if new_filename is None:
+            continue
+        mapped_template_files[new_filename] = template_file
+    if len(required_map) > 0:
         raise HTTPException(
             status_code=status.HTTP_400_BAD_REQUEST,
-            detail="No template was selected for rendering, template_output_name_mapping is empty.",
-        )
-
-    base_template = await template_service.get(id_or_identifier)
-
-    if not set(render_request.template_output_name_mapping.keys()).issubset(
-        base_template.template_files.keys()
-    ):
-        raise HTTPException(
-            status_code=status.HTTP_400_BAD_REQUEST,
-            detail="The required template files {} are not a subset of the template files for {}".format(
-                render_request.template_output_name_mapping.keys(), id_or_identifier
+            detail=(
+                f"The template files are missing required files: "
+                f"{required_map.keys()} for {id_or_identifier}"
```

Comment:
I think we can improve readability with something like:

```
    expected_keys = set(required_map.values())
    actual_keys = set(f.filename for f in base_template.template_files)
    if not expected_keys.issubset(actual_keys):
        raise HTTPException(
             status_code=status.HTTP_400_BAD_REQUEST,
             detail=(
                 f"The template files are missing required files: "
                 f"{expected_keys - actual_keys} for {id_or_identifier}"
```

Reply:
It's a little less efficient, but I agree that it's more readable. I also cleaned up the creation of the mapped_template_files a bit (with a walrus operator).

Reply:
Addressed in commit a59185b